### ISSUE TITLE
cleartext traffic

### ISSUE THREAD

**Issue by IzzySoft** [donmor/TiddloidLite#14]
Just wondering over what my scanner reported:

```
! repo/top.donmor.tiddloidlite_10.apk declares flag(s): usesCleartextTraffic
! repo/top.donmor.tiddloidlite_10.apk contains signature block blobs: 0x504b4453 (DEPENDENCY_INFO_BLOCK; GOOGLE)
```

What cleartext connections are used by the app? The `DEPENDENCY_INFO_BLOCK` can easily be avoided btw:

```gradle
android {
    dependenciesInfo {
        // Disables dependency metadata when building APKs.
        includeInApk = false
        // Disables dependency metadata when building Android App Bundles.
        includeInBundle = false
    }
}
```

For some background: that BLOB is supposed to be just a binary representation of your app's dependency tree. But as it's encrypted with a public key belonging to Google, only Google can read it – and nobody else can even verify what it really contains.

Thanks in advance!

**Comment by donmor** [comment]
The `usesCleartextTraffic` is there before Lite is forked from Tiddloid. It was added to support non-SSL connections(HTTP) in LAN. It may be removed someday since it is not necessary for Lite to make non-SSL connections.

As for `DEPENDENCY_INFO_BLOCK`, yeah I'd get rid of it somehow. P.S.: It's not clear if F-Droid builds have this🤔

**Comment by IzzySoft** [comment]
* `usesCleartextTraffic`: if it's not needed it's of course better removed. If it's still needed/used for LAN only, I can add that as exception, please let me know then. Err, I've just added that now – should you remove it my scanner will let me know.
* `DEPENDENCY_INFO_BLOCK`: thanks! And F-Droid.org will have that with reproducible builds, but only if you have it – as with RB they'd ship your build. But we're talking about [my repo](https://apt.izzysoft.de/fdroid) here, which always takes your APKs :wink:

I'll leave this issue open in case you need it for tracking `usesCleartextTraffic` – feel free to close if/when you don't (anymore).

**Comment by IzzySoft** [comment]
You're still working on the app, @donmor ?

**Comment by donmor** [comment]
> You're still working on the app, @donmor ?

Currently occupied, may have a look later. Sry😂

**Comment by IzzySoft** [comment]
OK, all fine – just wanted to check if I should drop the topic. Will keep it in the queue on my end then, and hope you don't mind a heads-up every few months or so 😉 All the best for whatever keeps you occupied!

**Comment by IzzySoft** [comment]
In case you're still on it: would you accept a PR shipping the metadata (app description + graphics) as used by IzzyOnDroid? We use the Fastlane standard – i.e. just some text + graphic files in a standardized directory structure, easy to maintain and no extra tooling needed. Ready to use by simply merging – and from then on enable you to define how your app is presented.

**Comment by donmor** [comment]
> In case you're still on it: would you accept a PR shipping the metadata (app description + graphics) as used by IzzyOnDroid? We use the Fastlane standard – i.e. just some text + graphic files in a standardized directory structure, easy to maintain and no extra tooling needed. Ready to use by simply merging – and from then on enable you to define how your app is presented.

Yup:-)

**Comment by IzzySoft** [comment]
Thanks! [Here it is](https://github.com/donmor/TiddloidLite/pull/15) then.

**Comment by donmor** [comment]
Merged. BTW does your repo accept triple-t? I have another app to submit to official repo, but my gitlab account is dead somehow.

**Comment by IzzySoft** [comment]
No, we do not support Triple-T here. Further, Triple-T has entered Maintenance mode, so you might wish to switch to Fastlane where you currently use Triple-T. For the `listings`, I could provide you a script relocating the files accordingly (it's in fact just moving some files around). The automation ("publisher") would be manual work AFAIK – but that one is outside of my expertise.

```bash
#!/bin/bash

FASTLANE=fastlane/metadata/android
mkdir -p "${FASTLANE}"
MOVE="cp"
FEATUREGRAPHIC="featured.png"
ICON="ic_launcher-playstore.png"

for dir in app/src/main/play/listings/*; do
  loc="$(basename $dir)"
  mkdir -p "${FASTLANE}/${loc}"
  [[ -f "${dir}/title.txt" && ! -f "${FASTLANE}/${loc}/title.txt" ]] && $MOVE "${dir}/title.txt" "${FASTLANE}/${loc}/title.txt"
  [[ -f "${dir}/short-description.txt" && ! -f "${FASTLANE}/${loc}/short_description.txt" ]] && $MOVE "${dir}/short-description.txt" "${FASTLANE}/${loc}/short_description.txt"
  [[ -f "${dir}/full-description.txt" && ! -f "${FASTLANE}/${loc}/full_description.txt" ]] && $MOVE "${dir}/full-description.txt" "${FASTLANE}/${loc}/full_description.txt"

  [[ -f "${dir}/graphics/feature-graphic/${FEATUREGRAPHIC}" && ! -f "${FASTLANE}/${loc}/images/featureGraphic.png" ]] && {
    mkdir -p "${FASTLANE}/${loc}/images"
    $MOVE "${dir}/graphics/feature-graphic/${FEATUREGRAPHIC}" "${FASTLANE}/${loc}/images/featureGraphic.png"
  }
  [[ -f "${dir}/graphics/icon/${ICON}" && ! -f "${FASTLANE}/${loc}/images/icon.png" ]] && {
    mkdir -p "${FASTLANE}/${loc}/images"
    $MOVE "${dir}/graphics/icon/${ICON}" "${FASTLANE}/${loc}/images/icon.png"
  }
  [[ -d "${dir}/graphics/phone-screenshots" && ! -d "${FASTLANE}/${loc}/images/phoneScreenshots" ]] && {
    mkdir -p "${FASTLANE}/${loc}//images/phoneScreenshots"
    $MOVE "${dir}/graphics/phone-screenshots/"* "${FASTLANE}/${loc}/images/phoneScreenshots/"
  }
done
```

> but my gitlab account is dead

For integration with IzzyOnDroid, you can simply ping me from your other project's issue tracker, I'll then check.
